cocos/2d/CCNode.h

```diff
@@ -1862,8 +1866,13 @@ class CC_DLL Node : public Ref
 #if CC_USE_PHYSICS
     PhysicsBody* _physicsBody;
 public:
-    void setPhysicsBody(Component* physicsBody) 
-    { 
+    void setPhysicsBody(PhysicsBody* physicsBody)
```

Comment:
If you really want Node not depend on physics you have to also remove `setPhysicsBody` method which was only added for backwards compatibility

If you don't call setPhysicsBody with a PhysicsBody the code crashes when using getPhysisBody method because _physicsBody is null in that case.

In conclusion I don't think changing Component to PhysicsBody here  breaks anyones code, as it makes no sense to call setPhysicsBody with a component other than PhysicsBody.

Hope that makes more sense now.